lang/cpp.go

```diff
@@ -247,188 +234,90 @@ func (c cpp) generatePrintCode(q *leetcode.QuestionData) string {
+}
+
+func (c cpp) Generate(q *leetcode.QuestionData) (*GenerateResult, error) {
+	filenameTmpl := getFilenameTemplate(q, c)
+	baseFilename, err := q.GetFormattedFilename(c.slug, filenameTmpl)
 	if err != nil {
-		return removeTempFiles(false, err)
+		return nil, err
 	}
-	inputs, outputs := c.parseGeneratedTestCases(string(filebuffer))
-
-	// execute
-	for caseId := 0; caseId < len(inputs); caseId++ {
-		// setup input & output
-		input := inputs[caseId]
-		expectedOutput := outputs[caseId]
-		if len(expectedOutput) == 0 {
-			expectedOutput = "[Not Implemented] to be retrieved from LC online testing..."
-		}
-
-		// setup command
-		timeLimit := 3000 * time.Millisecond
-		ctx, cancel := context.WithTimeout(context.Background(), timeLimit)
-		defer cancel()
-		cmd = exec.CommandContext(ctx, execFile, outputFile)
-		cmd.Stdin = strings.NewReader(input)
-		cmd.Stdout = os.Stdout
-		cmd.Stderr = os.Stderr
-
-		// execute test program
-		executeStart := time.Now()
-		err = cmd.Run()
-		elapsed := time.Since(executeStart)
-		if ctx.Err() != nil {
-			blue.Print("[TLE]")
-			fmt.Printf(" Case %d - Time Limit (%s) Exceeded! :(\n", caseId, timeLimit)
-			continue
-		}
-		if err != nil {
-			magenta.Print("[RE]")
-			fmt.Printf(" Case %d - Runtime Error! :(\n", caseId)
-			continue
-		}
-
-		// read test result
-		filebuffer, err := os.ReadFile(outputFile)
-		if err != nil {
-			return removeTempFiles(false, err)
-		}
-		actualOutput := string(filebuffer)
-
-		// judge & show result
-		dimCnt, returnType := c.getCppTypeName(q.MetaData.Return.Type)
-		if c.getJudgeResult(dimCnt, returnType, expectedOutput, actualOutput) {
-			green.Print("[AC]")
-			fmt.Printf(" Case %d - %s\n", caseId, elapsed)
-			fmt.Printf(" - Input:      %s\n", strings.ReplaceAll(input, "\n", "↩ "))
-			fmt.Printf(" - Output:     %s\n", actualOutput)
-		} else {
-			red.Print("[WA]")
-			fmt.Printf(" Case %d - %s\n", caseId, elapsed)
-			fmt.Printf(" - Input:      %s\n", strings.ReplaceAll(input, "\n", "↩ "))
-			fmt.Printf(" - Output:     %s\n", actualOutput)
-			fmt.Printf(" - Expected:   %s\n", expectedOutput)
-		}
+	genResult := &GenerateResult{
+		Question: q,
+		Lang:     c,
+		SubDir:   baseFilename,
 	}
 
-	return removeTempFiles(true, nil)
-}
-
-func (c cpp) Generate(q *leetcode.QuestionData) (*GenerateResult, error) {
+	separateDescriptionFile := separateDescriptionFile(c)
 	blocks := getBlocks(c)
 	modifiers, err := getModifiers(c, goBuiltinModifiers)
 	if err != nil {
 		return nil, err
 	}
-	codeContent, err := c.generateContent(q, blocks, modifiers)
+	codeFile, err := c.generateCodeFile(q, "solution.h", blocks, modifiers, separateDescriptionFile)
 	if err != nil {
 		return nil, err
 	}
-
-	testcaseStr := c.generateTestCases(q)
-	testContent := c.generateTest(q, testcaseStr)
-
-	filenameTmpl := getFilenameTemplate(q, c)
-	baseFilename, err := q.GetFormattedFilename(c.slug, filenameTmpl)
+	testFile, err := c.generateTestFile(q, "solution.cpp")
```

Comment:
What do you think of generating code and tests into the same file? Like what Go does now, this way only one single file is generated,  users can compile the single file themselves easier.

Reply:
> users can compile the single file themselves easier.

The "hardest" part for the user to compile is probably linking `LC_IO.h` anyways since it will be in a different folder.

I guess it's fine, but I think this proposal has a drawback. Due to the nature of C++, The auto-generated code file could be quite verbose and keeping them seperate could be better.

Reply:
I prefer to generate a single file, as the test code we generated is simple and not verbose in most cases.

Reply:
Ok, I will change this in the next commit.

Reply:
thank you very much!